python/mp/rosary.py

```diff
@@ -535,22 +548,38 @@ def mainloop(self, *args, **kwargs):
     ##########################################################################
     def fire_trigger(self, trigger_name, *args, **kwargs):
 
-        requested_trigger = self.trigger_registry.get(trigger_name)
-        # I really only expect there to be one trigger running at a time,
-        # but just in case, get everyone's names
-        running_trigger_names = [t.name for t in self.triggers]
+        # See if any currently running effects have hijacked this trigger
+        hijacked = self.trigger_hijacks.get(trigger_name)
+
+        if hijacked is not None and len(hijacked) > 0:
+
+            # I thought it'd be cool if the hijacks acted like a stack -
+            # if many effects hijack the same trigger, then only the newest
+            # effect's hijacked trigger fires
+            hijacked_effect, hijacked_method = hijacked[-1]
+
```

Comment:
That is the intention.

Although I've only tested 1 "layer" so far (if you consider what you're proposing as having 2 "layers")